python/pyarrow/src_arrow/flight.cc

```diff
@@ -371,24 +370,21 @@ Status CreateFlightInfo(const std::shared_ptr<arrow::Schema>& schema,
                         const std::vector<arrow::flight::FlightEndpoint>& endpoints,
                         int64_t total_records, int64_t total_bytes,
                         std::unique_ptr<arrow::flight::FlightInfo>* out) {
-  arrow::flight::FlightInfo::Data flight_data;
-  RETURN_NOT_OK(arrow::flight::internal::SchemaToString(*schema, &flight_data.schema));
-  flight_data.descriptor = descriptor;
-  flight_data.endpoints = endpoints;
-  flight_data.total_records = total_records;
-  flight_data.total_bytes = total_bytes;
-  arrow::flight::FlightInfo value(flight_data);
-  *out = std::unique_ptr<arrow::flight::FlightInfo>(new arrow::flight::FlightInfo(value));
+  auto result =
+    arrow::flight::FlightInfo::Make(*schema,
+                                    descriptor,
+                                    endpoints,
+                                    total_records,
+                                    total_bytes);
+  *out = std::unique_ptr<arrow::flight::FlightInfo>(&(*result));
```

Comment:
These should use ARROW_RETURN_NOT_OK

```suggestion
  ARROW_ASSIGN_OR_RAISE(auto result,
    arrow::flight::FlightInfo::Make(*schema,
                                    descriptor,
                                    endpoints,
                                    total_records,
                                    total_bytes));
  *out = std::unique_ptr<arrow::flight::FlightInfo>(std::move(result));
```

Reply:
If I use `std::move` here I get 

```cpp
/Users/alenkafrim/repos/arrow-2/python/pyarrow/src_arrow/flight.cc:379:10: error: no matching conversion for functional-style cast from 'typename remove_reference<FlightInfo &>::type' (aka 'arrow::flight::FlightInfo') to 'std::unique_ptr<arrow::flight::FlightInfo>'
  *out = std::unique_ptr<arrow::flight::FlightInfo>(std::move(result));
```

For time being I am keeping `&result` so that the code compiles. Can change it later.

Reply:
Oh sorry it should have been `std::unique_ptr<...>(new FlightInfo(std::move(result)))`

That also probably explains your crash! Since it currently creates a unique_ptr backed by a stack-allocated (temporary) FlightInfo

Reply:
Oh, haven't tried that one. Awesome, thank you very very much =)